Fabric.Authorization.AccessControl/src/app/services/global/services.service.ts

```diff
@@ -37,73 +37,167 @@ export class ServicesService {
+                    } else if (targetService === undefined) {
                         throw new Error(`The ${service.name} was not found in discovery service. Please ensure it is set up correctly.`);
+                    } else {
+                        service.url = targetService.ServiceUrl;
                     }
-                    service.url = targetService.ServiceUrl;
                 }
-
+                
                 return this.services.find(s => s.name === 'IdentityService').url;
             })
         );
     }
 
-    get identityServiceEndpoint(): string {
-        return this.services.find(s => s.name === 'IdentityService').url;
-    }
+    private findUrlBestMatch(url: string) {
+        var serviceUrlMatch = "";
+        this.services.forEach(element => {
+            if (element.url) {
+                var result = url.toLowerCase().includes(element.url.toLowerCase());
 
-    get authorizationServiceEndpoint(): string {
-        return this.services.find(s => s.name === 'AuthorizationService').url;
-    }
+                if (result) {
+                    if (element.url.length > serviceUrlMatch.length) {
+                        serviceUrlMatch = element.url.toLowerCase();
+                    }
+                }
+            }
+        });
 
-    get identityProviderSearchServiceEndpoint(): string {
-        return this.services.find(s => s.name === 'IdentityProviderSearchService').url;
+        var matchedService = this.services.find(s => s.url && s.url.toLowerCase() === serviceUrlMatch);
+        const serviceItem: IService = matchedService;
+        return serviceItem;
     }
 
-    get accessControlEndpoint(): string {
-        return this.services.find(s => s.name === 'AccessControl').url;
+    private trimRightChar(characters, char) {
+        var i = 0;
+        while (characters[characters.length - 1 - i] === char)
+            i++
+
+        return characters.substring(0, characters.length - i);
     }
 
-    public needsAuthToken(url: string) {
-        const urlLowerCase = url.toLowerCase();
-        const targetService: IService = this.services.find(s => urlLowerCase.includes(s.name.toLowerCase())) ? this.services.find(s => urlLowerCase.includes(s.name.toLowerCase())) : this.findUrlBestMatch(this.services, url);
-        return targetService ? targetService.requireAuthToken : false;
+    private buildServiceFilter(): string {
+        return this.services.map(service => `ServiceName eq \'${service.name}\'`).join(' or ');
     }
 
-    public findUrlBestMatch(services: IService[], url: string)
-    {
-       var serviceUrlMatch = "";
-       services.forEach(element => {
-        var result = url.toLowerCase().includes(element.url.toLowerCase());
-        if (result)
-        {
-          if (element.url.length > serviceUrlMatch.length)
-          {
-            serviceUrlMatch = element.url.toLowerCase();
-          }
+    private parseUrl(url) {
+        var location = document.createElement("a")
```

Comment:
Is there a simpler way to parse a url: https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
Not supported in IE 11.  This was the most browser friendly version.

Reply:
using URL would have been way easier